cpp/src/search/contains_nested.cu

```diff
@@ -63,4 +65,24 @@ bool contains_nested_element(column_view const& haystack,
            }) > 0;
 }
 
+std::unique_ptr<column> multi_contains_nested_elements(column_view const& haystack,
+                                                       column_view const& needles,
+                                                       rmm::cuda_stream_view stream,
+                                                       rmm::mr::device_memory_resource* mr)
+{
+  auto result_v = contains(table_view{{haystack}},
+                           table_view{{needles}},
+                           null_equality::EQUAL,
+                           nan_equality::ALL_EQUAL,
+                           stream,
+                           mr);
```

Comment:
Just curious why this detail function returns a `device_uvector` instead of a `column`?  
Will the result ever be larger than what a column supports?

Reply:
That `detail::contains(table_view, table_view)` function was initially refactored out of semi-join. Now, it is also used in several other places but the result is only used for temporary computation and never be returned to the user. Thus, `device_uvector`  is enough at this time.

It is still unclear whether we will expose this function to the public.

Reply:
I would also say that (especially for internal APIs like this, but occasionally even otherwise) I prefer using a strongly-typed object like a `device_uvector` rather than the type-erased (and nullable) `column`. We have more information available and convey it. Even for public APIs I like uvectors when we don't need type erasure or nullability because the returned values are clearly defined.

Reply:
This discussion is also related to #11356.